scipy/signal/_spectral_py.py

```diff
@@ -1987,14 +2025,29 @@ def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
 
     .. versionadded:: 0.16.0
     """
-    # Created sliding window view of array
+    xp = array_namespace(x)
+    x = xp.asarray(x)
+    # Created strided array of data segments
     if nperseg == 1 and noverlap == 0:
-        result = x[..., np.newaxis]
+        result = x[..., xp.newaxis]
     else:
         step = nperseg - noverlap
-        result = np.lib.stride_tricks.sliding_window_view(
-            x, window_shape=nperseg, axis=-1, writeable=True
-        )
+        # violating the array API standard gives us 4 orders of magnitude
+        # better performance via views/striding, so preserving some special
+        # casing
+        # see: https://github.com/data-apis/array-api/issues/641#issuecomment-1604884351
```

Comment:
I believe this branch used to have shims for array API support for libs other than CuPy and `torch` with something like the suggestion at: https://github.com/scipy/scipy/issues/18286#issuecomment-1522380436

I believe the underlying code also switched from `as_strided` to `sliding_windows_view` in `main` a few months ago as well, so this code path has mutated for the Nth time. Either way, this appears to pass the array API tests locally with the current skips I have at least, but doesn't really support the array API unless we restore a codepath that is a few orders of magnitude slower I think per the linked comment above.

Reply:
This looks pretty good. I'd only suggest moving `sliding_window_view` function to `scipy/_lib/_array_api.py`? Then it will work with a one-line change here, and it will work later in any other place we may want to use `sliding_window_view`.

Failing for now on non-numpy/cupy/torch and just adding a TODO comment seems okay.